src/common/code_environment.cc

```diff
@@ -79,6 +79,34 @@ int get_process_name(char *buf, int len)
   return 0;
 }
 
+#elif defined(_WIN32)
+
+int get_process_name(char *buf, int len)
+{
+  if (len <= 0) {
+    return -EINVAL;
+  }
+
+  int length = GetModuleFileNameA(nullptr, buf, len);
+  if (length <= 0)
+    return -ENOSYS;
+  if (length == len)
+    buf[length] = '\0';
```

Comment:
please note, per https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamea. the returned string is always nul terminated even if the buffer is too small.

> If the buffer is too small to hold the module name, the string is truncated to nSize characters including the terminating null character, the function returns nSize, and the function sets the last error to ERROR_INSUFFICIENT_BUFFER.

so no need to set the "\0" here.